### ISSUE TITLE
NextJS TypeError while disconnecting a wallet

### ISSUE THREAD

**Issue by Naggertooth** [airgap-it/beacon-sdk#367]
In the latest versions of package you have `TODO: type` or something like this while disconnecting a wallet
This encourages TypeError errors in NextJs

**Comment by AndreasGassmann** [comment]
Hi. I'm not sure what exactly you are referring to. The `TODO: type`s we have are code comments referring to incomplete typescript types, but that shouldn't be causing any errors in itself. Could you give me an example?

**Comment by Naggertooth** [comment]
I am already too far from this right now
May be later, if I would work around beacon
Thank you for replying!

ср, 3 авг. 2022 г., 10:57 AndreasGassmann ***@***.***>:

**Comment by RichAyotte** [comment]
I think the error that is being referred to is:
![image](https://user-images.githubusercontent.com/294987/204153114-4ae5ece3-6dab-4eee-9d21-28f539090d46.png)

Which is caused by the following found in DAppClient.ts:293:43)
```ts
          if (
            typedMessage.type === BeaconMessageType.Disconnect ||
            (message as any).typedMessage.type === BeaconMessageType.Disconnect // TODO: TYPE
          )
```

If `typedMessage` is any then can we add the optional chaining operator `typedMessage?.type`?

@AndreasGassmann let me know if you'd like a PR for this.

**Comment by IsaccoSordo** [comment]
Hi all,
this issue should be fixed in #551
